core/primitives/block_header.hpp

```diff
@@ -24,13 +25,13 @@ namespace kagome::primitives {
      * @param extrinsics_root extrinsics root
      * @param digest digest collection
      */
-    BlockHeader(Buffer parent_hash, size_t number, Buffer state_root,
-                Buffer extrinsics_root, Buffer digest);
+    BlockHeader(common::Hash256 parent_hash, size_t number, common::Hash256 state_root,
```

Comment:
don't forget to fix common::Hash256 encode/decode.
the same for std::array.
These types need to be considered as sequence of bytes one by one, not as collection, so number of items shouldn't be placed before bytes.